src/mysql/dialect.rs

```diff
@@ -0,0 +1,98 @@
+use std::sync::Arc;
+
+use arrow_schema::TimeUnit;
+use datafusion::{
+    common::HashMap,
+    logical_expr::Expr,
+    sql::{
+        sqlparser::ast,
+        unparser::{
+            dialect::{
+                DateFieldExtractStyle, Dialect, IntervalStyle, MySqlDialect as DFMySqlDialect,
+                ScalarFnToSqlHandler,
+            },
+            Unparser,
+        },
+    },
+};
+
+pub struct MySqlDialect {
```

Comment:
Should we move this into Datafusion crate to be consistent with DuckDB dialect?
https://github.com/apache/datafusion/blob/a256eb4e45d85701e9535c32eb84260c6817522a/datafusion/sql/src/unparser/dialect.rs#L365